cosmo_tester/test_suites/summary/__init__.py

```diff
@@ -0,0 +1,599 @@
+    'deployments': [
+        'blueprint_id',
+        'tenant_name',
+        'visibility',
+    ],
+    'executions': [
+        'status',
+        'blueprint_id',
+        'deployment_id',
+        'workflow_id',
+        'tenant_name',
+        'visibility',
+    ],
+    'nodes': [
+        'deployment_id',
+        'tenant_name',
+        'visibility',
+    ],
+    'node_instances': [
+        'deployment_id',
+        'node_id',
+        'state',
+        'host_id',
+        'tenant_name',
+        'visibility',
+    ],
+}
+
+
+@pytest.fixture(scope='module')
+def prepared_manager(manager, cfy, logger):
+    manager.client.snapshots.upload(
+        SUMMARY_SNAPSHOT_PATH,
+        'filtering_test_snapshot',
+    )
+    execution = manager.client.snapshots.restore(
+        'filtering_test_snapshot',
+        restore_certificates=False,
+        force=False,
+    )
+    try:
+        wait_for_execution(
+            manager,
+            execution,
+            logger)
+    except ExecutionFailed:
+        # See any errors
+        cfy.executions.list(['--include-system-workflows'])
+        raise
+    # Wait for any post restore commands to run
+    time.sleep(20)
+    yield manager
+
+
+class ExecutionWaiting(Exception):
+    """
+    raised by `wait_for_execution` if it should be retried
+    """
+    pass
+
+
+class ExecutionFailed(Exception):
+    """
+    raised by `wait_for_execution` if a bad state is reached
+    """
+    pass
+
+
+def retry_if_not_failed(exception):
+    return not isinstance(exception, ExecutionFailed)
+
+
+@retrying.retry(
+    stop_max_delay=5 * 60 * 1000,
+    wait_fixed=10000,
+    retry_on_exception=retry_if_not_failed,
+)
+def wait_for_execution(manager, execution, logger):
+    logger.info(
+        'Getting workflow execution [id={execution}]'.format(
+            execution=execution['id'],
+        )
+    )
+    execution = manager.client.executions.get(execution['id'])
+    logger.info('- execution.status = %s', execution.status)
+    if execution.status not in execution.END_STATES:
+        raise ExecutionWaiting(execution.status)
+    if execution.status != execution.TERMINATED:
+        logger.warning('Execution failed')
+        raise ExecutionFailed(
+            '{status}: {error}'.format(
+                status=execution.status,
+                error=execution['error'],
+            )
+        )
+    logger.info('Execution complete')
+    return execution
+
+
+def test_bad_field_selected(prepared_manager):
+    """
+        Confirm a helpful error message is returned if a bad field is selected
+    """
+    errors = []
+
+    for summary_endpoint in (
+        prepared_manager.client.summary.blueprints,
+        prepared_manager.client.summary.deployments,
+        prepared_manager.client.summary.executions,
+        prepared_manager.client.summary.nodes,
+        prepared_manager.client.summary.node_instances,
+    ):
+        try:
+            summary_endpoint.get(_target_field='notafield')
```

Comment:
perhaps this should use the CLI, so that part is also tested? system tests after all :P

Reply:
Added.